### DISCUSSION TITLE
Lazy initialization

### DISCUSSION THREAD

**Discussion by Korpyc** [klaviyo/klaviyo-swift-sdk#94] [Archive]
Described below also related to Android SDK, as I'm using both for my app written with Flutter.

In my case, there are no need in anonymous users at all. Only authorized. 
+ I'm sending to my backend server token key as well, so, I'm doing initialization only after success auth with known token & profile data.

My proposal:
1) Add optional "lazy" bool value inside "init with key" method or do lazy by default. This will prevent extra request to create anonymous profile.
SDK can do it with further interactions, such as event logging, profile updating, etc.

2) Add ability to setup profile token via custom property in "properties" section with profile data. This will also reduce redundant request in my case. 

With those updates, my app will need to make only one request, instead of three.

P.S. Please, fix current API for setting of profile. It should work also for updating email for next app starts as users can change their emails.

**Comment by ndurell** [comment]
Ahh got it! Potentially though there is a workaround. The anonymous profile should be merged with the non-anonymous profile when you set the email. If you use `setProfile` it will trigger a new anonymous profile but if you use `setEmail` the second will not reset the anonymous id and instead merge it into the email profile. Still I think you are right we could optimize the network traffic here so we don't make two requests.

**Comment by ndurell** [comment]
Also of note in an upcoming release we will update the push token in a new way which can include creating the profile. So potentially we will be able to get your use case down to 1 api call. My hope is that we can make similar improvements for events as well in the future as those apis become available.

**Comment by evan-masseau** [comment]
FYI we're closing discussions in our SDK in favor of using issues for both bugs and feature requests to streamline our feedback channels.
I think Noah's answered your question, but if you do have further feedback, feel free to open an issue!
